### PR TITLE
VAGOV-6046: Adding click event handlers to multiple elements.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
Adds multiple click event handlers to various elements.
See: https://github.com/department-of-veterans-affairs/va.gov-team/issues/1846#issuecomment-534157534

### CODE REVIEW

**Comment by rianfowler** [comment]
Please tag the FE review group in a comment after this has a peer review approval. Thanks!

File: src/platform/site-wide/user-nav/components/HelpMenu.jsx
Comment (ncksllvn) [inline src/platform/site-wide/user-nav/components/HelpMenu.jsx]: Can we switch to `recordEvent(..)` here instead? It's just an alias to `window.data.Layer.push`

Comment (ethanteague) [reply src/platform/site-wide/user-nav/components/HelpMenu.jsx]: You bet, @ncksllvn - updated :+1:

**Review by rianfowler — APPROVED** [review]

File: src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx
Comment (ncksllvn) [inline src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx]: Can we import and use that `recordEvent` function here too? 

`recordEvent` is also globally available in templates as well, so it should always work as an alias.

Comment (ethanteague) [reply src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx]: Np, @ncksllvn - updated.

File: src/platform/site-wide/user-nav/components/HelpMenu.jsx
Comment (ncksllvn) [inline src/platform/site-wide/user-nav/components/HelpMenu.jsx]: Hey @ethanteague , I think this should have been wrapped in a function, like -

```
 onClick={() => recordEvent({ event: 'nav-jumplink-click' })}
```

Same with https://github.com/department-of-veterans-affairs/vets-website/pull/10818/files#diff-6ce07a2b29c8f4123001e03206daf27eR58.

Currently, it's called `recordEvent` immediately  (when the component loads) instead of when the event actually fires.

**Pull request merged**
